### PR TITLE
gpg-agent: add missing options

### PR DESCRIPTION
**Pull request by rvolosatovs**

### CODE REVIEW

File: modules/services/gpg-agent.nix [hunk 1]
```diff
@@ -29,10 +29,44 @@ in
...
+
+      disableScDaemon = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Do not make use of the scdaemon tool. This option has the effect of disabling the ability to do smartcard operations.
+        '';
+      };
+
+      writeEnvFile = mkOption {
```
Comment (rycee) [inline modules/services/gpg-agent.nix]: Are you sure this is needed? From what I can tell this option is no longer available:
```
WARNING: "--write-env-file" is an obsolete option - it has no effect
```

File: modules/services/gpg-agent.nix [hunk 2]
```diff
@@ -29,10 +29,44 @@ in
...
       enableSshSupport = mkOption {
         type = types.bool;
         default = false;
-        description = "Whether to use the GnuPG key agent for SSH keys.";
+        description = ''
+          Whether to use the GnuPG key agent for SSH keys.
+        '';
+      };
+
+      noGrab = mkOption {
```
Comment (rycee) [inline modules/services/gpg-agent.nix]: I'd suggest to instead have `grabKeyboardAndMouse = mkOption { … }` defaulting to `true`. When possible I'd prefer avoiding negated options for pedagogical reasons. Can have `optional (!cfg.grabKeyboardAndMouse) "no-grab"` below.

File: modules/services/gpg-agent.nix [hunk 3]
```diff
@@ -29,10 +29,44 @@ in
...
+        type = types.bool;
+        default = false;
+        description = ''
+          Tell the pinentry not to grab the keyboard and mouse. This option should in general not be used to avoid X-sniffing attacks.
+        '';
+      };
+
+      disableScDaemon = mkOption {
```
Comment (rycee) [inline modules/services/gpg-agent.nix]: Same here, nicer with `enableScDaemon` that defaults to `true`.

**Comment by rycee** [comment]
Thanks for the contribution! I've added a few minor comments.

File: modules/services/gpg-agent.nix (hunk 1 above)
Comment (rvolosatovs) [reply modules/services/gpg-agent.nix]: I made these changes a while ago, could be it got deprecated.

**Comment by rycee** [comment]
Looking good :-) I squashed and rebased to master in c023b0532a7c9761840aefc0e059b2e424fb1520.

**Pull request closed without merging**